### PR TITLE
Taida Century NRF52 minidev board variants

### PR DESCRIPTION
**Pull request by micooke**
Regarding issue #200 - add pin mapping and associated documentation for RGZ and RSM variants

Also fix incorrect defines in boards.txt -DNRF51_S132 => -DNRF52_S132

### CODE REVIEW

**Comment by dlabun** [comment]
I take it from your last post that this is not ready for merging?

**Comment by micooke** [comment]
It should be ready, #6a3b6c5 should have fixed that.

**Comment by micooke** [comment]
I don't know what your thoughts on solderable links are in general as they are on a few of the boards (DK, microbit, taida).

In this instance i piggyback off  ```lfclk_flags```, but if you wanted this capability for other boards i should add a separate compile parameter - maybe ```variant_flags```?

File: boards.txt [hunk 1]
```diff
@@ -248,9 +248,18 @@ STCT_nRF52_minidev.menu.softdevice.s132=S132
 STCT_nRF52_minidev.menu.softdevice.s132.softdevice=s132
 STCT_nRF52_minidev.menu.softdevice.s132.softdeviceversion=2.0.1
 STCT_nRF52_minidev.menu.softdevice.s132.upload.maximum_size=409600
-STCT_nRF52_minidev.menu.softdevice.s132.build.extra_flags=-DNRF52 -DS132 -DNRF51_S132
+STCT_nRF52_minidev.menu.softdevice.s132.build.extra_flags=-DNRF52 -DS132 -DNRF52_S132
 STCT_nRF52_minidev.menu.softdevice.s132.build.ldscript=armgcc_s132_nrf52832_xxaa.ld
 
+STCT_nRF52_minidev.menu.board_variant.none=None
+STCT_nRF52_minidev.menu.board_variant.rgz.build.lfclk_flags=-DUSE_LFXO -DSTCT_NRF52_minidev
```
Comment (sandeepmistry) [inline boards.txt]: Why are the `-DSTCT_NRF52_minidev`, ` -DSTCT_NRF52_minidev_RGZ`, and `-DSTCT_NRF52_minidev_RSM` part of `build.lfclk_flags`?

Comment (micooke) [reply boards.txt]: Yeah I made mention of this in https://github.com/sandeepmistry/arduino-nRF5/issues/200#issuecomment-335063303

Its a bit of a hack. It really needs a ```build.variant_flags```, and i didnt know if you wanted to add this just for one use case.

Comment (sandeepmistry) [reply boards.txt]: I think the `build.variant_flags` is cleaner.

**Comment by micooke** [comment]
Ill try and update this shortly

**Comment by sandeepmistry** [comment]
Closing this for now due to lack of activity. Please let us know if/when this is ready for review and testing.

**Pull request closed without merging**